crime-applications-adaptor/build.gradle

```diff
@@ -1,149 +1,196 @@
 plugins {
-	id 'java'
-	id 'jacoco'
-	id 'org.springframework.boot' version '3.2.4'
-	id 'io.spring.dependency-management' version '1.1.4'
-	id "org.sonarqube" version "5.0.0.4638"
-	id 'info.solidsoft.pitest' version '1.15.0'
-	id 'org.jsonschema2dataclass' version '6.0.0'
+    id 'java'
+    id 'jacoco'
+    id 'org.springframework.boot' version '3.2.4'
+    id 'io.spring.dependency-management' version '1.1.4'
+    id "org.sonarqube" version "5.0.0.4638"
+    id 'info.solidsoft.pitest' version '1.15.0'
+    id 'org.jsonschema2dataclass' version '6.0.0'
 }
 
 group = 'uk.gov.justice.laa.crime'
 
 def versions = [
-		springdocVersion       : "2.5.0",
-		okhttpVersion          : "4.9.3",
-		mockwebserverVersion   : "4.12.0",
-		pitest                 : "1.15.8",
-		commonsLang3Version    : "3.14.0",
-		postgresqlVersion      : "42.3.3",
-		springauthserverVersion: "1.0.1",
-		festVersion            : '1.4',
-		harmcrestVersion       : '1.3',
-		jjwtVersion            : '0.12.5',
-		validationapiVersion   : '2.0.1.Final',
-		resilience4jVersion    : "2.2.0",
-		commonsioVersion       : '2.11.0',
-		sentryVersion 		   : "7.6.0"
+        springdocVersion         : "2.5.0",
+        okhttpVersion            : "4.9.3",
+        mockwebserverVersion     : "4.12.0",
+        pitest                   : "1.15.8",
+        commonsLang3Version      : "3.14.0",
+        postgresqlVersion        : "42.3.3",
+        springauthserverVersion  : "1.0.1",
+        festVersion              : '1.4',
+        harmcrestVersion         : '1.3',
+        jjwtVersion              : '0.12.5',
+        validationapiVersion     : '2.0.1.Final',
+        resilience4jVersion      : "2.2.0",
+        commonsioVersion         : '2.11.0',
+        sentryVersion            : "7.6.0",
+        serenityVersion          : "4.1.6",
+        jUnitPlatformSuiteVersion: "1.10.2",
+        cucumberEngineVersion    : "7.16.1",
```

Comment:
I would consider having the functional api test as a separate (child) Gradle module with it's own `build.gradle`, as it will result in simpler, more focused Gradle build files for both the application and the api tests.
Which IMHO should make the approach easier to understand and when modification is needed, less disruption to the application `build.gradle`.

I would consider simplifying the module name to `api-test` as it follows the same convention used elsewhere (with hyphen separators), as i'm not sure what additional information the 'functional' prefix communicates.

Reply:
This was my initial thought but I took the easy route to avoid mass refactoring, however, now that you've mentioned it... new commits incoming...!